### DISCUSSION TITLE
Can 2 Separate DLRS roll up to the same Field on the same Object?

### DISCUSSION THREAD

**Discussion by zack0816** [SFDO-Community/declarative-lookup-rollup-summaries#1265] [Q&A]
We are attempting to create a summary rollup of total Assets Under Management (AUM) in the Financial Service Cloud (FSC). We need two DLRS configurations to run that roll up to the same field on the same object. One for Household Accounts. Another for Business Accounts. After hours of tinkering, this seems impossible. Am I correct in my assessment? 

Here is how we have it configured (I took additional relationship criteria out for simplicity, but DLRS appears to be ignoring any other filter I throw at it):

Household - Total AUM Primary Owner

Parent Obj: Account
Child Obj: FinServ__FinancialAccount__c
Relationship Field: FinServ__Household__c
**Relationship Criteria**: FinServ__Household__c != NULL 
Field to Aggregate: FinServ__Balance__c
Aggregate Operation: Sum
Aggregate Result Field: FinServ__TotalAUMPrimaryOwner__c
Calculation Mode: Realtime

Business - Total AUM Primary Owner

Parent Obj: Account
Child Obj: FinServ__FinancialAccount__c
Relationship Field: FinServ__Household__c
**Relationship Criteria**: FinServ__Household__c = NULL
Field to Aggregate: FinServ__Balance__c
Aggregate Operation: Sum
Aggregate Result Field: FinServ__TotalAUMPrimaryOwner__c
Calculation Mode: Realtime

The above configuration has the Household AUM updated as expected, and the Business account incorrectly sets AUM to $0 and vice versa depending on which: 

- DLRS was active at the time
- FinServ__FinancialAccount__c was updated  

**It appears DLRS isn't designed for multiple Summaries that Rollup to the same field on the same object. Does anyone know precisely why that is?** 

Note: We've explored activating and deactivating the competing DLRS configurations with the Process Builder Calculation Mode. AUMs of the Business or Household accounts were still zeroed out even after ensuring the DLRS configurations were not simultaneously active.  

FYI @jasonniebauer

**Comment by stephenbrown1** [comment]
hmm you may need to put some logic in the calling trigger around when to run one vs the other, and set the calc mode of both to Developer. Or you can use Process Builder and call it via PB logic if you don't want to go down the editing trigger logic path.
(/side note for DLRS collabs: need to update calc option to add Flow also...)
more info here: https://sfdo-community-sprints.github.io/DLRS-Documentation/Architecture/calculates.html#important-note

**Reply by zack0816** [reply]
Thank you for the response and link to the documentation, @stephenbrown1.

**Reply by zack0816** [reply]
Upon further testing, I was incorrect. Not only does it zero out the AUM of the opposite account type when both DRLS configurations are on, it zeros out the AUM of the opposite account type if you update its financial account's balance when the other is active.

For example, if Household - Total AUM Primary Owner is active and you update the one FinServ__Balance__c of a business account, the AUM of the business account goes to 0. The opposite is also true.
